avoid shipping sysv init script in case systemd is detected

manner82 opened the pull request:
Some linux distros ship by default with broken systemd <-> sysv compatibility. On these, if the init.d script is also present, it prevents the systemd service to get enabled. Observed the problem at least on SLES 15 SP3, Suse Tumbleweed, Suse Leap. (On the other side, RHEL, debian do work correctly.)

The goal of this patch is to remove the sysv init script when the systemd service gets enabled.

Example error on SLES:
```
> sudo systemctl enable myservice
Synchronizing state of myservice.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install enable myservice
/sbin/insserv: No such file or directory
```

On Suse:
```
# systemctl enable myservice
Synchronizing state of vasd.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install enable myservice
ln -sf ../vasd /etc/init.d/rc2.d/S50myservice
ln: failed to create symbolic link '/etc/init.d/rc2.d/S50myservice': No such file or directory
```

manner82 committed:
packaging/pp: fix systemctl service enable on SLES 15 SP3

On suse/SLES enabling the service fails, because systemd
chooses to use the init.d script over its own service config and
sysv compatibility is broken on a default install.

manner82 committed:
pp.back.systemd.svc: stop the service on removal

Previously this was done by the sysv script, but now we have removed that,
so it fails silently.

manner82 committed:
pp.back.systemd: check status and restart with systemctl

If the service is not running according to its sysv script (eg.
systemd service is not using a pidfile the sysv script needs), or
its sysv script is not present any more (the next upgrade), we also
check how systemd thinks, and stop with that.

We also try to restart the service in case something went wrong with it
(if it is in a failed state).

millert reviewed (approved):
My only concern is that this removes a packaged file listed in %files but that doesn't seem to be an issue in practice.  It doesn't look like we need to do this for .deb files; the Debian systemd-sysv-install  script appears to just update the rc.d links.

manner82 committed:
rpm: do not warn for missing init.d script during package removal

Adds a %config(missingok) in the %files section in front of the sysv
init script, because now we might remove it during install (if systemd
is detected).

millert reviewed (approved):
Looks good to me.
